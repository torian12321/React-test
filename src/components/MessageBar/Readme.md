#### Basic
Similar to <Message> but less relevant
```
<MessageBar>Content</MessageBar>
```

#### Types
```
<div>
  <MessageBar type='info'>Info content</MessageBar>
  <MessageBar type='success'>Success content</MessageBar>
  <MessageBar type='danger'>Danger content</MessageBar>
  <MessageBar type='warning'>Warning content</MessageBar>
</div>
```
