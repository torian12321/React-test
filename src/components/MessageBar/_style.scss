@import '../../styles/conf/index.scss';

// Vars
$messageBar-color-border: $themeColor_1 !default;
$messageBar-color-bg    : darken($white, 5) !default;

// Block
.messageBar{
  display    : block;
  margin     : 20px 0;
  padding    : 8px 12px;
  border-left: $messageBar-color-border solid 4px;
  background : $messageBar-color-bg;
  @include smart-text-color($messageBar-color-bg);
  
  // Modifiers
  @each $stateName, $stateColor in $states {
    &--#{$stateName} {
      border-color: nth($stateColor, 1);
    }
  }
}
