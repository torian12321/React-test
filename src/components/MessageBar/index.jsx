import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import './_style.scss';

class MessageBar extends React.Component {
  render(){
      return(
        <div className={classNames(
          'messageBar',
          this.props.type && `messageBar--${this.props.type}`,
          this.props.className,
        )}>
          {this.props.children}
        </div>
    );
  }
}

MessageBar.propTypes = {
  children : PropTypes.any.isRequired,
  className: PropTypes.string,
  type     : PropTypes.oneOf(['info', 'success', 'danger', 'warning'])
};

module.exports = MessageBar;
