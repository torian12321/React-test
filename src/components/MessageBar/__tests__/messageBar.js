import React      from 'react';
import renderer   from 'react-test-renderer';
import MessageBar from '../index';

describe('component', () => {
  describe('<MessageBar />', () => {

    it('should render correctly', () => {
      expect(renderer.create(
        <MessageBar>Lorem Ipsum</MessageBar>
      )).toMatchSnapshot();
    });

    it('Should render danger type', () => {
      expect(renderer.create(
        <MessageBar type='danger'>Lorem Ipsum</MessageBar>
      )).toMatchSnapshot();
    });

    it('Should render important state', () => {
      expect(renderer.create(
        <MessageBar important>Lorem Ipsum</MessageBar>
      )).toMatchSnapshot();
    });

  });
});
