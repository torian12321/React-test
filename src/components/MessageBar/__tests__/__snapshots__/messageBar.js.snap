// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <MessageBar /> Should render danger type 1`] = `
<div
  className="messageBar messageBar--danger"
>
  Lorem Ipsum
</div>
`;

exports[`component <MessageBar /> Should render important state 1`] = `
<div
  className="messageBar"
>
  Lorem Ipsum
</div>
`;

exports[`component <MessageBar /> should render correctly 1`] = `
<div
  className="messageBar"
>
  Lorem Ipsum
</div>
`;
